main.go

```diff
@@ -418,10 +421,8 @@ func setupManager(options ctrl.Options) (ctrl.Manager, error) {
 		return nil, err
 	}
 
-	if options.MetricsBindAddress != "0" {
-		if err := mgr.AddMetricsExtraHandler("/server-metrics", promhttp.Handler()); err != nil {
-			return nil, err
-		}
+	if options.Metrics.BindAddress != "0" {
+		options.Metrics.ExtraHandlers = map[string]http.Handler{"/server-metrics": promhttp.Handler()}
```

Comment:
Defining extra metric handlers looks to have moved to this option instead:
- https://github.com/kubernetes-sigs/controller-runtime/blob/5f8d96b043ef8ccadc544535415395be336630e8/pkg/metrics/server/server.go#L69-L74